packages/react/src/index.js

```diff
@@ -153,10 +121,42 @@ microlink.io/${error.code.toLowerCase()}
 
       setLoading(false)
     },
-    [mediaProps, setData]
+    [updateState, mediaProps, setData]
   )
 
-  useEffect(toFetchData, [url, setData, hasIntersected])
+  const toFetchData = useCallback(() => {
+    if (canFetchData) {
+      setLoading(true)
+
+      const fetch = fetchData
+        ? fetchFromApi(apiUrl, apiUrlProps)
+        : Promise.resolve({})
+
+      fetch
+        .then(({ data }) => mergeData(data))
+        .catch(error => {
+          setLoading(false)
+          setIsError(true)
+          console.error(`
+┌───────────────┐
+│ Microlink SDK │
+└───────────────┘
+
+${error.description}
+
+${JSON.stringify(error.data)}
+
+id   ${error.headers['x-request-id']}
+uri  ${error.url}
+code ${error.code} (${error.statusCode})
+
+microlink.io/${error.code.toLowerCase()}
+`)
+        })
+    }
+  }, [apiUrlProps, fetchData, apiUrl, mergeData, canFetchData])
+
+  useEffect(toFetchData, [toFetchData, url, setData, hasIntersected])
```

Comment:
Este es un poco sensible... entiendo que no pasa nada. Pero es raro añadir como dependencia el propio "toFetchData", ya que el propio "toFetchData" cambiará según las dependencias la línea 157.

Pero veo que todas las dependencias tiran de useMemo o useCallback, así que en principio diría que sí es de fiar. 👍

Reply:
Hey @midudev

>  es raro añadir como dependencia el propio "toFetchData"

Realmente no creo que es raro. Si estamos siguiendo las _rules of hooks_, al ser `toFetchData` un método ubicado a nivel del body del component, debemos incluirlo como dependency si lo vamos a consumir dentro del `useEffect`.

Lo que si es algo extraño y posiblemente confuso es el syntax, que estamos haciendo esta especie especie de shorthand en lugar de:

```
useEffect(() => {
  toFetchData()
}, [...])
```

@Kikobeats no recuerdo si esto fui yo, pero por mi lo podríamos ajustar para ser más _readable_ en vez de ganar un par de LOC 😛.

> Pero veo que todas las dependencias tiran de useMemo o useCallback

Yep! Y las que no, vienen de props - las cuales deben hacer que hagamos un _re-run_ del effect de todas formas.